var/spack/repos/builtin/packages/py-matplotlib/package.py

```diff
@@ -23,57 +23,61 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA
 ##############################################################################
 from spack import *
-import os
+
 
 class PyMatplotlib(Package):
-    """Python plotting package."""
+    """matplotlib is a python 2D plotting library which produces publication
+    quality figures in a variety of hardcopy formats and interactive
+    environments across platforms."""
+
     homepage = "https://pypi.python.org/pypi/matplotlib"
     url      = "https://pypi.python.org/packages/source/m/matplotlib/matplotlib-1.4.2.tar.gz"
 
+    version('1.5.1', 'f51847d8692cb63df64cd0bd0304fd20')
     version('1.4.2', '7d22efb6cce475025733c50487bd8898')
     version('1.4.3', '86af2e3e3c61849ac7576a6f5ca44267')
 
     variant('gui', default=False, description='Enable GUI')
-    variant('ipython', default=False, description='Enable ipython support')
-
-    extends('python', ignore=r'bin/nosetests.*$|bin/pbr$')
 
-    depends_on('py-pyside', when='+gui')
-    depends_on('py-ipython', when='+ipython')
+    # Required dependencies
+    extends('python@2.7:2.8,3.4:3.5', ignore=r'bin/nosetests.*$|bin/pbr$')
+    depends_on('py-numpy@1.6:')
+    depends_on('py-setuptools')
+    depends_on('py-dateutil@1.1:')
     depends_on('py-pyparsing')
-    depends_on('py-six')
-    depends_on('py-dateutil')
+    depends_on('libpng@1.2:')
     depends_on('py-pytz')
-    depends_on('py-nose')
-    depends_on('py-numpy')
-    depends_on('py-mock')
-    depends_on('py-pbr')
-    depends_on('py-funcsigs')
+    depends_on('freetype@2.3:')
+    depends_on('py-cycler@0.9:')
+    # depends_on('py-tornado')
+
+    # Optional GUI framework
+    depends_on('tk@8.3:',       when='+gui')
+    depends_on('py-pyqt@4.0:',  when='+gui')
+    depends_on('pygtk@2.4:',    when='+gui')
+    depends_on('wxPython@2.8:', when='+gui')
```

Comment:
Haha, that's why `+gui` is off by default. But I can see why this makes you cringe. According to the [backend documentation](http://matplotlib.org/faq/usage_faq.html#what-is-a-backend), it looks like you're right. I think you can choose any one of these to use as your backend.

What do you think we should do? We could split each of these into a separate variant. I'm not sure whether they are build dependencies or runtime dependencies, but from the previous file filtering in the package, I have a feeling that they need to be present and be set in that config file. This makes it sound like they are mutually exclusive.